tests/test_subscribe.py

```diff
@@ -0,0 +1,159 @@
+"""Tests for the subscription registry and http server module."""
```

Comment:
I haven't looked closely at the tests yet - but it does pass for me as well locally.